API-GUIDELINE.md

```diff
@@ -0,0 +1,254 @@
+# API GUIDELINES
+
+## Overview
+As commercetools is 100% API First, it's important to have a well designed developer experience.
+Additionally we aim to achieve more automation and improve the review process of new changes in the APIs.
+
+With this in mind, we have created "API Guidelines" that aims to help developers to design APIs:
+
+ - simple
+ - consistent 
+ - self-explanatory
+ 
+The Keywords to have a good Guidelines are:
+ 
+ - Consistency => following the Guidelines
+ - Uniformity => checking in the code if there are other properties with similar name and adjust in base of it or improve it
+ - Explicability => explain with a few words the purpose of the property avoiding abbreviations (there are some [exceptions](#exceptions)).
+
+## Scope
+The purpose of these guidelines is to achieve the definition of consistent practices and patterns for all API endpoints.
+
+## URI
+Our API URIs are composed by:
+   
+    Base Uri like:
+        https://api.{region}.{cloudProvider}.commercetools.com 
+
+    {projectKey}
+
+    resources
+
+    by ID
+
+    by external ID with equals e.g. customers/key={key}
+
+See our documentation for more details https://docs.commercetools.com/http-api.html#hosts
+
+## Resources
+Resource is an object or representation of something, which has some associated data with it and can be helped by methods to operate on it.
+
+A resource can be an entity or a collection:
+ - Entity is a resource that has to be in singular.
+ - Collections are a set of resources that must be in plural.
+ 
+A resource may contain sub-collection resources.
+
+## Methods
+The HTTP verbs used are;
+
+ - GET -> retrieve data
+ - POST -> create or update data
```

Comment:
IMO it's worth mentioning that a create should return a `201`, whereas an update should return a `200` HTTP status code.

Reply:
it's just because there is people that use GET for everything ;)

Reply:
I can fix it ;)